tests/foreman/maintain/test_clone.py

```diff
@@ -0,0 +1,190 @@
+"""Tests for satellite-maintain clone
+
+:Requirement: foreman-maintain
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: SatelliteClone
+
+:Assignee: lpramuk
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import pytest
+import requests
+
+from robottelo import constants
+from robottelo.config import settings
+from robottelo.hosts import Satellite
+
+SSH_PASS = settings.server.ssh_password
+
+
+@pytest.mark.parametrize("sat_ready_rhel", [8], indirect=True)
+@pytest.mark.parametrize('backup_type', ['online', 'offline'])
+def test_positive_clone_backup(target_sat, sat_ready_rhel, backup_type):
+    """Make online/offline backup of satellite and clon it (restore it).
+
+    :id: 5b9182d5-6789-4d2c-bcc3-6641b96ab277
+
+    :steps:
+        1. Create online/offline backup
+        2. Copy it to target rhel
+        3. Install satellite-clone
+        4. Run satellite clone on target rhel
+
+    :expectedresult:
+        1. Satellite-clone should be successfull
+    """
+    rhel_version = sat_ready_rhel._v_major
+
+    # SATELLITE PART - SOURCE SERVER
+    # Enabling and starting services
+    assert target_sat.cli.Service.enable().status == 0
+    assert target_sat.cli.Service.start().status == 0
+    # Doing backup
+    backup_result = target_sat.cli.Backup.run_backup(
+        backup_dir='/var/backup',
+        backup_type=backup_type,
+        options={'assumeyes': True, 'plaintext': True},
+    )
+    assert backup_result.status == 0
+    # Getting backup directory
+    sat_backup_dir = backup_result.stdout.strip().split()[-2]
+
+    # Copying satellite backup to target RHEL
+    assert (
+        target_sat.execute(
+            f'''sshpass -p "{SSH_PASS}" scp -o StrictHostKeyChecking=no -r {sat_backup_dir} \
+            root@{sat_ready_rhel.hostname}:/backup/'''
+        ).status
+        == 0
+    )
+
+    # RHEL PART - TARGET SERVER
+    assert sat_ready_rhel.execute('ls /backup/config_files.tar.gz').status == 0
+    # Registering to cdn
+    sat_ready_rhel.register_to_cdn()
+    # Disabling repositories
+    assert sat_ready_rhel.execute('subscription-manager repos --disable=*').status == 0
+    # Getting satellite maintenance repo
+    res = requests.get(
```

Comment:
would `target_sat.dogfood_repofile` not work in this case?

Reply:
It would not as `sat_ready_rhel` is `ContentHost`, not `Satellite` object.
If `dogfood_repofile` is used on ContentHost, then the client repo is defined instead of the Satellite repo.

Reply:
The suggestion was more to use the target_sat to get the correct repofile URL. However, a previous comment expressed a need for even greater control. I have agreed to moving this to a method.

Reply:
Resolving this thread, let's discuss it in https://github.com/SatelliteQE/robottelo/pull/9986#discussion_r952389184